Implement ext-foreign-toplevel-list

wineee opened the pull request.

wineee committed:
Implement ext-foreign-toplevel-list

zccrs commented on src/server/protocols/wextforeigntoplevellistv1.cpp:
```diff
@@ -0,0 +1,118 @@
+        }
+
+        wlr_ext_foreign_toplevel_handle_v1_state state = {
+            .title = surface->title().toUtf8().constData(),
```
这里的 toUtf8().constData 只是一个临时的对象，执行完毕应该会被销毁的，后续的 .title 和 .app_id 就是个无效的内容了。

zccrs commented on src/server/protocols/wextforeigntoplevellistv1.cpp:
```diff
@@ -0,0 +1,118 @@
+    return nativeInterface<qw_ext_foreign_toplevel_list_v1>()->handle()->global;
+}
+
+WAYLIB_SERVER_END_NAMESPACE 
```
尾部加一个空行

zccrs commented on src/server/protocols/wextforeigntoplevellistv1.h:
```diff
@@ -0,0 +1,38 @@
+    wl_global *global() const override;
+};
+
+WAYLIB_SERVER_END_NAMESPACE 
```
尾部加个换行

wineee committed:
fix: Fix string lifetime issues in multiple files

- Fix WForeignToplevel string lifetime issues in titleChanged and appIdChanged callbacks
- Fix WSeat name string lifetime issue in create() method
- Fix WSocket lockFile string lifetime issue in open() call

These fixes prevent potential crashes caused by using temporary QByteArray
objects whose data becomes invalid after the expression ends.

zccrs commented on src/server/protocols/wforeigntoplevelv1.cpp:
```diff
@@ -154,8 +156,10 @@ class Q_DECL_HIDDEN WForeignToplevelPrivate : public WObjectPrivate
-        handle->set_title(surface->title().toUtf8());
-        handle->set_app_id(surface->appId().toLocal8Bit());
+        const auto title = surface->title().toUtf8();
+        const auto appId = surface->appId().toLocal8Bit();
```
toLatin1
